lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -644,4 +670,153 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    @Override
+    public StatusEvent updateEvent(final StatusEvent ev)
+            throws IndexingException,
+                   RetriableIndexingException {
+
+        // brute force get latest state and update event as events
+        // can be played out of order in the case of failed events being replayed.
+
+        StatusEventType eventType = ev.getEventType();
+        String accessGrpId = ev.getAccessGroupId().get().toString();
+        String objectId = ev.getAccessGroupObjectId().get();
+
+        // update a rename event
+        if (eventType.equals(StatusEventType.RENAME_ALL_VERSIONS)) {
+            // check if name has changed and update state of lastestName
+
+            try {
+                GetObjectInfo3Results objInfo =
+                        getObjectInfo(Long.decode(accessGrpId), Long.decode(objectId));
+                String latestName = objInfo.getInfos().get(0).getE2();
+
+                if( ev.getNewName().equals(latestName) )
+                    return ev;
+                else {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(), eventType).
```

Comment:
We discussed this. Given that the call is being made in WorkspaceEventHandler the event must contain the WS objid.

Reply:
From what I recall we talked about the fact that it must contain the workspace id / access group id, not the object ID. There are definitely events that don't include an object ID like PUBLISH_ACCESS_GROUP (or something like that)

Reply:
How do I get the object info to check if the object has been renamed if I don't have the object id?

Reply:
Looking at where this method is actually used, event expansion has already occurred and so this method in practice would never see access group (e.g. workspace) level events (which won't have object IDs).

So I think there are two options: 1) throw an exception if it's an access group level event, or 2) if it's an access group level event, don't set the object name.

this would be easier to deal with if we added constraints to the events and documented them.

Reply:
OK. I added a filter to only update object level events.